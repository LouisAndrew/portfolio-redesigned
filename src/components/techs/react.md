---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
