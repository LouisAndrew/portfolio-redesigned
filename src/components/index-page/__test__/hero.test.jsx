import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import { LocationProvider } from '@reach/router'

import Hero from '../hero'

describe('Hero component on landing page.', () => {
        
        afterEach( cleanup )
        
        const mainText = 'Hello'
        const parText = 'Test paragraph'

        const renderWithRouter = ui => <LocationProvider>{ui}</LocationProvider>

        const theme = { center: () => `` }
        const renderWithStyledTheme = ui => <ThemeProvider theme={theme}>{ui}</ThemeProvider>

        it('renders without crashing', () => {

                const div = document.createElement('div')
                ReactDOM.render(renderWithRouter( renderWithStyledTheme(<Hero />) ), div )
        })

        it('renders correctly', () => {

                const { getByTestId } = render( renderWithRouter( renderWithStyledTheme(<Hero heading={mainText} subheading={parText} />) ) )

                expect( getByTestId('hero-heading') ).toHaveTextContent(mainText)
                expect( getByTestId('hero-par') ).toHaveTextContent(parText)
        })

        it('matches snapshot', () => {
                
                const tree = renderer.create( renderWithRouter( renderWithStyledTheme( <Hero heading={mainText} subheading={parText} /> ) ) ).toJSON()

                expect(tree).toMatchSnapshot()
        })
})
