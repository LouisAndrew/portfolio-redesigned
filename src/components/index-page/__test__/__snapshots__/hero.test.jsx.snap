// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component on landing page. matches snapshot 1`] = `
<section
  className="sc-AxirZ fxMaFa wrap"
  id="particles-js"
>
  <div
    className="sc-AxhCb evnbtu"
  >
    <h1
      data-testid="hero-heading"
      id="hero-heading"
    >
      Hello
    </h1>
    <p
      data-testid="hero-par"
    >
      Test paragraph
    </p>
    <button
      className="sc-AxjAm gqzgEY"
      color="ft"
      data-testid="button"
      onClick={[Function]}
    />
  </div>
</section>
`;
